sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2Suites.scala

```diff
@@ -654,6 +654,14 @@ class HiveThriftBinaryServerSuite extends HiveThriftJdbcTest {
       assert(resultSet.getString(1) === "4.56")
     }
   }
+
+  test("SPARK-28463: Thriftserver throws BigDecimal incompatible with HiveDecimal") {
```

Comment:
That also indicates our thrift server might have various issues we might not catch. 

Could you bulid a test framework that can directly re-run all the tests in SQLQueryTestSuite via Thrift Server? We do not need to run them always, since they could be very time consuming.

Reply:
https://issues.apache.org/jira/browse/SPARK-28527